### PR TITLE
Edit typo in jax.linen.softmax()

### PR DESCRIPTION
**Pull request by posgnu**
## Description

I got the error while implementing the transformer using this module.

keyword argument in `nn.softmax()` was misspecified.

Fix `dim` to `axis`.

### CODE REVIEW

**Comment by murphyk** [comment]
Yes, it seems the new code matches the linen API: https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.log_softmax.html#flax.linen.log_softmax
